lib/libc++/Makefile

```diff
@@ -77,6 +82,11 @@ CXXFLAGS+=	-std=c++11
 .endif
 
 LIBADD+=	cxxrt
+.ifdef LIBCHERI
+
+# XXXAR: why is this needed?
+LDFLAGS+=	-L../libcxxrt
```

Comment:
I suspect `libcxxrt` isn't being built early enough due to a lack of additions to the hardcoded build order in the buildcheri target in `src/Makefile.inc1`.  I'm not totally convinced of that theory though.